### PR TITLE
Provide Additional context in README

### LINKED ISSUE #30: Experience Report
🎉   **Everything worked and was able to resolve issues _self-directed_**
🙇  Thanks for anyone that contributed to this! 🙌 

In previous roles, I've contributed to onboarding. In my last role, I was doing this in PowerShell & Windows 10. Having some captured record from new team members was helpful for me to know what they ran into so that if I was not available to assist the individual serving _as guide_, I would be able to consider the issue and attempt to address it (thankfully, `laptop.zsh` doesn't have to be concerned with the registry 😱  and deal with numerous **reboots**)

I wanted to include error text and some contextual text so that is someone in the future is assisting in a "tech mentor" or Corvus Guide 🐦 🗺️   - they might search the Issues here and the workarounds (or actions) might unblock them (this is why there is _way more_ text would seem _necessary_ 🙇 ).

#### `cask` issue within _bundle_

The command, [`cask "graphiql"`](https://github.com/corvusinsurance/laptop/blob/main/laptop.zsh#L71), caused the following error:
```
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-cask /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask` exited with 128.
Installing graphiql has failed!
Installing postgresql
Homebrew Bundle failed! 1 Brewfile dependency failed to install.
failed
```

This is something that Greg @gjastrab ran into and noted here: 
- https://github.com/corvusinsurance/laptop/pull/27#discussion_r537771915
  - Found that via [Machine Setup](https://corvusinsurance.atlassian.net/wiki/spaces/PM/pages/44304085/Machine+Setup) wiki page (mentioning `asdf`)
 
##### :gear: Workaround _**(allegedly)**_ 
I did _confirm_ that [the commenter here](https://github.com/corvusinsurance/laptop/pull/27#discussion_r539391807). 

I'm a _might-bit_ ashamed that I do not know the brew checkout I had wasn't seeming applying the `cask` installer, as noted above. I had a fresh OS install, so it would seem like the first execution of the `./laptop.zsh` would have gotten this version. 

A las, but the time I noticed this, I had already begun the process of "double checking" and I had done the following:
```
brew update
brew doctor
brew config
```
After this, I was able to run `./laptop.zsh`, *clean-green* - no snags. 

⚠️ Re-running the `bundle` command might have a negative side-effect ... 🔍  investigate (potentially)
> In mentioning this to Greg, he mentioned that `brew bundle` run [in the script](https://github.com/corvusinsurance/laptop/blob/main/laptop.zsh#L56-L73) might clobber existing formulas.

<details> 

## Output Context 

Summary: `cask` not found

```
Updating Homebrew formulae ...
Already up-to-date.
==> Tapping homebrew/bundle
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-bundle'...
remote: Enumerating objects: 5704, done.
remote: Total 5704 (delta 0), reused 0 (delta 0), pack-reused 5704
Receiving objects: 100% (5704/5704), 1.29 MiB | 833.00 KiB/s, done.
Resolving deltas: 100% (3339/3339), done.
Tapped 1 command (108 files, 1.7MB).
Tapping homebrew/services
Tapping hashicorp/tap
Installing ctags
Installing git
Installing openssl
Installing awscli
Installing pre-commit
Installing gh
Installing hashicorp/tap/vault
==> Tapping homebrew/cask
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask'...
error: unable to create file Casks/ubar.rb: Interrupted system call
Updating files: 100% (3905/3905), done.
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'

Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-cask /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask` exited with 128.
Installing graphiql has failed!
Installing postgresql
Homebrew Bundle failed! 1 Brewfile dependency failed to install.
failed
```

##### :gear: Action(s) Taken

```
andrewlenards@Andrews-MacBook-Pro asdf % brew link p11-kit
Linking /usr/local/Cellar/p11-kit/0.23.22... 8 symlinks created.
andrewlenards@Andrews-MacBook-Pro asdf % 
```

```
andrewlenards@Andrews-MacBook-Pro asdf % sudo brew services start unbound
Password:
Warning: Taking root:admin ownership of some unbound paths:
  /usr/local/Cellar/unbound/1.13.0/sbin
  /usr/local/Cellar/unbound/1.13.0/sbin/unbound
  /usr/local/opt/unbound
  /usr/local/opt/unbound/sbin
  /usr/local/var/homebrew/linked/unbound
This will require manual removal of these paths using `sudo rm` on
brew upgrade/reinstall/uninstall.
==> Successfully started `unbound` (label: homebrew.mxcl.unbound)
```

</details>

### Within `asdf/install.zsh` 

The following two mentioned more as "I did the suggested" **action**, it's a nicety of `brew` that they'll offer up the text in the first place. 

**p11-kit**
```
==> Installing gnupg dependency: p11-kit
==> Pouring p11-kit-0.23.22.big_sur.bottle.tar.gz
Error: The `brew link` step did not complete successfully
```

:gear: **action taken**
```
# ran following mentioned in the more verbose example in Details below 
brew link p11-kit 
```

**unbound**
```
==> Installing gnupg dependency: unbound
==> Pouring unbound-1.13.0.big_sur.bottle.tar.gz
Error: Failed to read Mach-O binary: /usr/local/Cellar/unbound/1.13.0/share/man/man8/unbound-anchor.8
```

:gear: **action taken**
```
# ran following mentioned in the more verbose example in Details below 
sudo brew services start unbound
```

<details>

## Error 1 

```
==> Installing gnupg dependency: p11-kit
==> Pouring p11-kit-0.23.22.big_sur.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Interrupted system call @ rb_file_s_symlink - (../../../../Cellar/libidn2/2.3.0/share/gtk-doc/html/libidn2/right-insensitive.png, /usr/local/share/gtk-doc/html/libidn2/right-insensitive.png)

You can try again using:
  brew link p11-kit
==> Summary
🍺  /usr/local/Cellar/p11-kit/0.23.22: 63 files, 3MB
==> Installing gnupg dependency: libevent
==> Pouring libevent-2.1.12.big_sur.bottle.tar.gz
🍺  /usr/local/Cellar/libevent/2.1.12: 57 files, 2MB
==> Installing gnupg dependency: unbound
==> Pouring unbound-1.13.0.big_sur.bottle.tar.gz
Error: Failed to read Mach-O binary: /usr/local/Cellar/unbound/1.13.0/share/man/man8/unbound-anchor.8
==> Caveats
To have launchd start unbound now and restart at startup:
  sudo brew services start unbound
```

## Error 2 

```
==> gnutls
If you are going to use the Guile bindings you will need to add the following
to your .bashrc or equivalent in order for Guile to find the TLS certificates
database:
  export GUILE_TLS_CERTIFICATE_DIRECTORY=/usr/local/etc/gnutls/
gpg: directory '/Users/andrewlenards/.gnupg' created
gpg: keybox '/Users/andrewlenards/.gnupg/asdf-nodejs.gpg' created
gpg: keyserver receive failed: No route to host
gpg: /Users/andrewlenards/.gnupg/trustdb.gpg: trustdb created
gpg: key D7062848A1AB005C: public key "Beth Griggs <bgriggs@redhat.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keyserver receive failed: No route to host
gpg: key 7434390BDBE9B9C5: public key "Colin Ihrig <cjihrig@gmail.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keyserver receive failed: No route to host
gpg: key D3A89613643B6201: public key "Danielle Adams <adamzdanielle@gmail.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keyserver receive failed: No route to host
gpg: key C97EC7A07EDE3FC1: public key "keybase.io/jasnell <jasnell@keybase.io>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keyserver receive failed: No route to host
gpg: key 770F7A9A5AE15600: public key "Michaël Zasso (Targos) <targos@protonmail.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keyserver receive failed: No route to host
gpg: key E73BC641CC11F4C8: public key "Myles Borins <myles.borins@gmail.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keyserver receive failed: No route to host
gpg: key C43CEC45C17AB93C: public key "Richard Lau <rlau@redhat.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keyserver receive failed: No route to host
gpg: key C273792F7D83545D: public key "Rod Vagg <rod@vagg.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keyserver receive failed: No route to host
gpg: key F07496B3EB3C1762: public key "Ruben Bridgewater <ruben@bridgewater.de>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keyserver receive failed: No route to host
gpg: key 97B01419BD92F80A: public key "Ruy Adorno <ruyadorno@hotmail.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keyserver receive failed: No route to host
gpg: key F13993A75599653C: public key "Shelley Vohr (security is major key) <shelley.vohr@gmail.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1

Elm Setup…

Rust Setup…

Python Setup…

👨‍🔬asdf successfully configured! Next step in the Readme 👩‍🔬

```

</details> 

### Within `asdf/elm.zsh`

Written up as Issue #29 (more context available in the issue).

The `elm.zsh` will have 3 failures in attempting to install the global npm tools `elm`, `elm-format`, `elm-oracle`:
```
Installing Elm…
Yarn requires Node.js 4.0 or higher to be installed.
Yarn requires Node.js 4.0 or higher to be installed.
Yarn requires Node.js 4.0 or higher to be installed.
```

<details>

## `asdf current` Output

```
andrewlenards@Andrews-MacBook-Pro asdf % asdf current 
elixir          1.11.2          /Users/andrewlenards/repos/laptop/asdf/.tool-versions
elm             0.19.1          /Users/andrewlenards/repos/laptop/asdf/.tool-versions
erlang          23.1.1          /Users/andrewlenards/repos/laptop/asdf/.tool-versions
nodejs          ______          No version set. Run "asdf <global|shell|local> nodejs <version>"
python          3.7.9 2.7.16    /Users/andrewlenards/repos/laptop/asdf/.tool-versions
rust            1.47.0          /Users/andrewlenards/repos/laptop/asdf/.tool-versions
yarn            1.22.10         /Users/andrewlenards/repos/laptop/asdf/.tool-versions
andrewlenards@Andrews-MacBook-Pro asdf % 
```

## Example output

```
andrewlenards@Andrews-MacBook-Pro asdf % zsh ./elm.zsh


Installing Elm…
Yarn requires Node.js 4.0 or higher to be installed.
Yarn requires Node.js 4.0 or higher to be installed.
Yarn requires Node.js 4.0 or higher to be installed.

👨‍🔬Elm successfully installed! Next step in the Readme 👩‍🔬
andrewlenards@Andrews-MacBook-Pro asdf % 
```

</details>

#### Fetch Failed
The `curl` was creating a zero-byte zip for the VS Code application: 

```
laptop % ls -lha  ~/Downloads 
total 1150080
-rw-r--r--@  1 andrewlenards  staff     0B Feb  8 16:23 vscode.zip
andrewlenards@Andrews-MacBook-Pro laptop % 
```

##### :gear: Workaround

Used the information provided in the script output to download the zip file from the specified link. Installed VSCode.app, and then re-ran `./vs-code/install.zsh`. _Rough_ steps offered in Details below.

<details>

## :gear: Workaround Steps

- Manually downloaded from the FWLINK: https://go.microsoft.com/fwlink/?LinkID=620882
- Did `unzip` … to /Applications 
- Launch via Spotlight 
- Ran CMD SHIFT P ‘install “code”’ to add
- ⚠️ this doesn't modify `.zshrc`, so `code` will be missing from `$PATH` zsh session when it refreshes

```
andrewlenards@Andrews-MacBook-Pro laptop % zsh ./vs-code/install.zsh
🌀🌀 Installing Visual Studio Code 🌀🌀
🌀 Installing Visual Studio Code
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Archive:  /Users/andrewlenards/Downloads/vscode.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /Users/andrewlenards/Downloads/vscode.zip or
        /Users/andrewlenards/Downloads/vscode.zip.zip, and cannot find /Users/andrewlenards/Downloads/vscode.zip.ZIP, period.
The file /Applications/Visual Studio Code.app does not exist.
🌀 If I failed to unzip, then download manually from here - https://go.microsoft.com/fwlink/?LinkID=620882
🌀 Add the 'code' cli manually then rerun this script
🌀 In VS-Code, run pallete command (cmd + shift + p) -> type 'Shell Command'
andrewlenards@Andrews-MacBook-Pro laptop % ls -lha  ~/Downloads 
total 1150080
-rw-r--r--@  1 andrewlenards  staff     0B Feb  8 16:23 vscode.zip
andrewlenards@Andrews-MacBook-Pro laptop % 
```

</details> 

#### Settings & Snippets not found 
This is likely caused by me executing `./vs-code/install.zsh` from the _root_ of `laptop`, but the files were not found extensions & snippets: 
```
 laptop % zsh ./vs-code/install.zsh
🌀🌀 Installing Visual Studio Code 🌀🌀
🌀 Looks like the CLI installed correctly
./vs-code/install.zsh:16: no such file or directory: /Users/andrewlenards/repos/laptop/extensions.config
🌀 Copy my settings into lib/app/code
cp: /Users/andrewlenards/repos/laptop/settings.config: No such file or directory
🌀 Copy my snipets into lib/app/code
cp: directory /Users/andrewlenards/Library/Application Support/Code/User/snippets does not exist
🌀🌀 Visual Studio is lookin' Good! 🌀🌀
```

##### :gear: Workaround 
```
 laptop % cd vs-code 
 vs-code % zsh ./install.zsh 
```

<details> 

## Output From Workaround 

```
 laptop % cd vs-code 
 vs-code % zsh ./install.zsh 
🌀🌀 Installing Visual Studio Code 🌀🌀
🌀 Looks like the CLI installed correctly
Installing extensions...
Extension 'peterjausovec.vscode-docker' not found.
Make sure you use the full extension ID, including the publisher, e.g.: ms-dotnettools.csharp
Failed Installing Extensions: peterjausovec.vscode-docker
Installing extensions...
Installing extension 'mjmcloug.vscode-elixir' v1.1.0...
(node:28874) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Extension 'mjmcloug.vscode-elixir' v1.1.0 was successfully installed.
Installing extensions...
Installing extension 'mrmlnc.vscode-scss' v0.9.0...
(node:29033) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Extension 'mrmlnc.vscode-scss' v0.9.0 was successfully installed.
Installing extensions...
Extension 'robertohuertasm.vscode-icons' not found.
Make sure you use the full extension ID, including the publisher, e.g.: ms-dotnettools.csharp
Failed Installing Extensions: robertohuertasm.vscode-icons
Installing extensions...
Installing extension 'sbrink.elm' v0.26.0...
(node:29352) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Extension 'sbrink.elm' v0.26.0 was successfully installed.
🌀 Copy my settings into lib/app/code
🌀 Copy my snipets into lib/app/code
cp: directory /Users/andrewlenards/Library/Application Support/Code/User/snippets does not exist
🌀🌀 Visual Studio is lookin' Good! 🌀🌀
```

A _further_ workaround is to do:
```
# from `./laptop/vs-code`
mkdir ~/Library/Application\ Support/Code/User/snippets/
cp ./snippets/* ~/Library/Application\ Support/Code/User/snippets/
```
</details> 

#### Docker extension not found 

From the **_Details_** of the above section, there is output to show that `peterjausovec.vscode-docker` is no longer in the VS Code Extension Marketplace
```
Extension 'peterjausovec.vscode-docker' not found.
```

### README Edits

- [x] Mention assumed actions before start
    - [x] ssh-keygen / profile (link to GitHub Help)
    - [x] will need `git` installed to clone 
              - macOS is _awesome_ enough to prompt to install the developer tools
              - only reason to mention this is _'time budgeting'_ for someone that might have meetings/etc 
- [x] Mention shifting to following `asdf/README.md` after `laptop.zsh` complete
       - It's not _really_ needed, the script is directing to follow the `asdf/README.md` instructions. Depends on if someone is nervous and _rushing_ that might miss it (but easily recoverable through reaching out to their **Corvus Guide**). 
- [x] Add guidance on doing `git config --global` on new machine
       - (I assume lots of folks had a `dotfiles` repo they'll bring into a machine and this is handled by that personalization)

The script outputs following:
```
Next cd into the asdf directory and follow the installation instructions there
andrewlenards@Andrews-MacBook-Pro laptop % ls
README.md	asdf		atom		laptop.zsh	shell-helpers	vault_aws.zsh	vs-code
andrewlenards@Andrews-MacBook-Pro laptop % cd asdf
andrewlenards@Andrews-MacBook-Pro asdf % ls
README.md	elm.zsh		install.zsh	plugins.zsh
andrewlenards@Andrews-MacBook-Pro asdf % 
```

### PR DESCRIPTION
**Pull request by alenards**
Resolves #30. 

This work address the **_"Edits"_** section of Issue #30. It does not handle the changes that would likely be desirable to address the VS Code snags that were encounter. That will be handled in forthcoming work/effort.

_**Seeking reviewers...**_ - Greg / @gjastrab or Graham / @gmcintire - if you have others that want to review, can you consider assigning them _as well_? (it looked like you both were recent contributors; thought that was a good place to begin)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +54,30 @@ Optional: Atom Editor
...
+To make sure you're contributions are consistently attributed to you, you can use `git config` to set that up:
+```
+# Example Corvid
+git config --global user.name "Geococcyx Californianus"
+git config --global user.email greater_roadrunner@common-names.taxon
+```
+
+If you're not comfortable with `vi` or `vim`, you'll also want to set your default editor:
+```
+git config --global core.editor "code --wait"
```
Comment (gjastrab) [inline README.md:68]: 😍 Perfect timing - we will be mentioning this in the CDC tomorrow! 🙌

File: README.md [hunk 2]
```diff
@@ -2,6 +2,20 @@
 
 Corvus version of the thoughtbot laptop script
 
+:bird: Related **Aviary** page: [Machine Setup](https://corvusinsurance.atlassian.net/l/c/18dLhzPz)
+
+## Assumptions
```
Comment (gjastrab) [inline README.md:7]: Awesome, verifying the base state before installation is a great addition!

**Review by gjastrab — APPROVED** [review]
Thanks @alenards this is awesome!

**Comment by alenards** [comment]
I'm gonna squash & merge this. If I need to wait for 2 approvals, I'll ask forgiveness 🙇

**Pull request merged**